utils/s3/client.cc

```diff
@@ -237,37 +237,43 @@ future<> map_s3_client_exception(std::exception_ptr ex) {
 
 }
 
-future<> client::make_request(http::request req, http::experimental::client::reply_handler handle, http::reply::status_type expected) {
-    authorize(req);
-    auto& gc = find_or_create_client();
-    return gc.http.make_request(std::move(req), std::move(handle), expected).handle_exception([] (auto ex) {
+future<> client::do_make_request(group_client& gc, http::request req, http::experimental::client::reply_handler handle, std::optional<http::reply::status_type> expected_opt, seastar::abort_source* as) {
+    auto expected = expected_opt.value_or(http::reply::status_type::ok);
```

Comment:
Why is this conversion needed? Why not just pass expected_opt as it down to gc.http.make_request() it accepts optional<status_type> naturally

Reply:
Mainly because http::result::status_type::ok is not clearly specified as the default when nullopt is provided. In fact, if arg to http client is nullopt  the result of the call is ignored.